src/EditorFeatures/Core/ExtractMethod/ExtractMethodCommandHandler.cs

```diff
@@ -28,136 +28,138 @@
+        IBackgroundWorkIndicatorContext waitContext)
+    {
+        _threadingContext.ThrowIfNotOnUIThread();
 
-            var cancellationToken = waitContext.UserCancellationToken;
+        var cancellationToken = waitContext.UserCancellationToken;
 
-            var document = await textBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChangesAsync(waitContext).ConfigureAwait(false);
-            if (document is null)
-                return;
+        var document = await textBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChangesAsync(waitContext).ConfigureAwait(false);
+        if (document is null)
+            return;
 
-            var options = await document.GetExtractMethodGenerationOptionsAsync(_globalOptions, cancellationToken).ConfigureAwait(false);
-            var result = await ExtractMethodService.ExtractMethodAsync(
-                document, span, localFunction: false, options, cancellationToken).ConfigureAwait(false);
-            Contract.ThrowIfNull(result);
+        var options = await document.GetExtractMethodGenerationOptionsAsync(_globalOptions, cancellationToken).ConfigureAwait(false);
+        var result = await ExtractMethodService.ExtractMethodAsync(
+            document, span, localFunction: false, options, cancellationToken).ConfigureAwait(false);
+        Contract.ThrowIfNull(result);
 
-            result = await NotifyUserIfNecessaryAsync(
-                document, span, options, result, cancellationToken).ConfigureAwait(false);
-            if (result is null)
-                return;
+        result = await NotifyUserIfNecessaryAsync(
+            document, span, options, result, cancellationToken).ConfigureAwait(false);
+        if (result is null)
+            return;
 
-            var cleanupOptions = await document.GetCodeCleanupOptionsAsync(_globalOptions, cancellationToken).ConfigureAwait(false);
-            var (formattedDocument, methodNameAtInvocation) = await result.GetFormattedDocumentAsync(cleanupOptions, cancellationToken).ConfigureAwait(false);
-            var changes = await formattedDocument.GetTextChangesAsync(document, cancellationToken).ConfigureAwait(false);
+        var cleanupOptions = await document.GetCodeCleanupOptionsAsync(_globalOptions, cancellationToken).ConfigureAwait(false);
+        var (formattedDocument, methodNameAtInvocation) = await result.GetFormattedDocumentAsync(cleanupOptions, cancellationToken).ConfigureAwait(false);
+        var changes = await formattedDocument.GetTextChangesAsync(document, cancellationToken).ConfigureAwait(false);
 
-            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
+        await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(cancellationToken);
 
-            ApplyChange_OnUIThread(textBuffer, changes, waitContext);
+        ApplyChange_OnUIThread(textBuffer, changes, waitContext);
 
+        if (methodNameAtInvocation != default)
```

Comment:
while this should not be possible, we don't technically know if we can refind the name token for the invocation call after extract method processes.

in the old code we would do everything up front, and bail out of the entire operation if we coudln't find this.  Now, we presume we'll succeed, and if we can't find the method call for some strange reason, we simply do not start the rename session afterwards.